modin/engines/base/frame/data.py

```diff
@@ -329,9 +329,9 @@ def _filter_empties(self):
         self._column_widths_cache = [w for w in self._column_widths if w != 0]
         self._row_lengths_cache = [r for r in self._row_lengths if r != 0]
 
-    def _apply_index_objs(self, axis=None):
+    def synchronize_labels(self, axis=None):
         """
-        Apply the index object for specific `axis` to the `self._partitions` lazily.
+        Synchronize labels by applying the index object for specific `axis` to the `self._partitions` lazily.
```

Comment:
```suggestion
        Synchronize labels by applying the `self.index` object for specific `axis` to the `self._partitions` lazily.
```
?

Reply:
`self.index` corresponds to labels by axis=0. When we say just index object, we imply pandas.Index to be applied to specific axis. I suggest keeping the docstring as it is.

Reply:
+1 I think we can leave it as it is. Is that ok with you @vnlitvinov?

Reply:
Uhh, I missed "for the specific `axis`" part :) it's fine as it is then.